notebooks/MAST/astroqueryMAST/beginner_search_astroqueryMAST/beginner_search_astroqueryMAST.ipynb

```diff
@@ -0,0 +1,395 @@
+    "\n",
+    "Since we already have a list of observations, we can use that as the starting point for our query. To keep it simple, let's look at only the last observation from our search above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's select a small subset from our critera search above\n",
+    "newObsList = obsByCriteria[3:]\n",
+    "\n",
+    "# Now we get the list of products associated with that observation\n",
+    "dataProducts = Observations.get_product_list(newObsList)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "dataProducts.show_in_notebook(display_length=5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Filtering the Data Products\n",
+    "\n",
+    "After the data products have been retrieved, you can use `filter_products` to download only data products that meet your given criteria. Available filters are listed [here](https://mast.stsci.edu/api/v0/_productsfields.html). Some examples are: “mrp_only” (Minimum Recommended Products) and “extension” (file extension).\n",
+    "\n",
+    "Note: the ‘AND’ operation is performed for a list of filters, and the ‘OR’ operation is performed within a filter set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scienceProducts = Observations.filter_products(dataProducts, productType=[\"SCIENCE\"],\n",
+    "                                               mrp_only=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "scienceProducts.show_in_notebook(display_length=5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Downloading Products\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In additon to pre-filtering, `download_products` also allows you to filter data as you request the download. In this instance, we will only download the drizzled files (drz.fits).\n",
+    "\n",
+    "Products will by default be downloaded into the current working directory, in a subdirectory called \"mastDownload.\"<br>\n",
+    "The full local file paths will have the form \"mastDownload/Mission/Observation ID/file.\"\n",
+    "\n",
+    "Note: `download_products` includes caching by default, so if the files have been downloaded, they will not be downloaded again unless caching is turned off."
```

Comment:
I'm not 100% sure if this means a file won't be downloaded if the contents changed but the name has not.  If so, we will want to warn users that if they suspect a new version of the same file is availbale, they should turn off the cache (and ideally we should confirm this works as-intended in astroquery).  If not, they will want to delete their astroquery.mast cache (which I'd rather not get into within a beginner guide notebook!)